network/p2p/GENERAL_RULES.md

```diff
@@ -0,0 +1,69 @@
+# General message validations
+
+## Message reactions
+
+Node can react to a message in three ways:
+- `Accept` - message is valid, accepted and processed further.
+- `Ignore` - message is invalid, ignored and not processed further, violation count is increased.
+- `Reject` - message is rejected, ignored and not processed further, violation count is set to violation threshold, peer is banned for the current round.
+
+## Semantic assertions
+
+- Message validator maintains a state for each validator.
+- Generally, if violation happens by a small margin, the message is ignored. 
+  - If an assertion has a rule for ignoring, the rule has higher priority than this one.
+- Generally, If violation happens by a large margin, the message is rejected. 
+  - If an assertion has a rule for rejecting, the rule has higher priority than this one.
+- If violation count reaches a certain threshold, all further messages from the validator for current round are rejected.
+- Each round resets violation count to 0.
+- Validator attests only once per epoch.
+  - TODO: Need to check if upon reorg there can be more than one duty per epoch.
+  - First violation is ignored. Further ones are rejected.
+- Message round is equal to estimated current round.
+- Violation by [1; ceil(MAX_POSSIBLE_ROUND * 0.2)] rounds is ignored. Violations above that are rejected.
```

Comment:
- future rounds should be punished
- earlier ones are to be discussed
- test if clock drift may cause any false positives (if so, consider checking Prysm code regarding clock drift)